src/table.ts

```diff
@@ -1861,16 +1869,19 @@ class Table extends common.ServiceObject {
       throw new Error('You must provide at least 1 row to be inserted.');
     }
 
-    const json = extend(true, {}, options, {
-      rows,
-    });
+    const json = extend(true, {}, options, {rows});
 
     if (!options.raw) {
       json.rows = rows.map((row: RowMetadata) => {
-        return {
-          insertId: uuid.v4(),
-          json: Table.encodeValue_(row),
+        const encoded: InsertRow = {
+          json: Table.encodeValue_(row)!,
         };
+
+        if (options.createInsertId !== false) {
```

Comment:
A few lines below, “json.raw” is deleted. Maybe this option should be removed from the object that is sent upstream as well?

Reply:
Good call! Just added a commit for that